src/main/java/redis/clients/jedis/DefaultJedisSocketConfig.java

```diff
@@ -0,0 +1,163 @@
+    return connectionTimeout;
+  }
+
+  @Override
+  public int getSoTimeout() {
+    return soTimeout;
+  }
+
+  @Override
+  public boolean isSSL() {
+    return ssl;
+  }
+
+  @Override
+  public SSLSocketFactory getSSLSocketFactory() {
+    return sslSocketFactory;
+  }
+
+  @Override
+  public SSLParameters getSSLParameters() {
+    return sslParameters;
+  }
+
+  @Override
+  public HostnameVerifier getHostnameVerifier() {
+    return hostnameVerifier;
+  }
+
+  @Override
+  public HostAndPortMapper getHostAndPortMapper() {
+    return hostAndPortMapper;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private int connectionTimeout = Protocol.DEFAULT_TIMEOUT;
+    private int soTimeout = Protocol.DEFAULT_TIMEOUT;
+
+    private boolean ssl = false;
+    private SSLSocketFactory sslSocketFactory = null;
+    private SSLParameters sslParameters = null;
+    private HostnameVerifier hostnameVerifier = null;
+
+    private HostAndPortMapper hostAndPortMapper = null;
+
+    private Builder() {
+    }
+
+    public DefaultJedisSocketConfig build() {
+      return new DefaultJedisSocketConfig(connectionTimeout, soTimeout, ssl,
+          sslSocketFactory, sslParameters, hostnameVerifier, hostAndPortMapper);
+    }
+
+    public Builder withTimeout(int timeout) {
+      return withConnectionTimeout(timeout).withSoTimeout(timeout);
+    }
+
+    public Builder withConnectionTimeout(int connectionTimeout) {
+      this.connectionTimeout = connectionTimeout;
+      return this;
+    }
+
+    public Builder withSoTimeout(int soTimeout) {
+      this.soTimeout = soTimeout;
+      return this;
+    }
+
+    public Builder withSsl(boolean ssl) {
+      this.ssl = ssl;
+      return this;
+    }
+
+    public Builder withSslSocketFactory(SSLSocketFactory sslSocketFactory) {
+      this.sslSocketFactory = sslSocketFactory;
+      return this;
+    }
+
+    public Builder withSslParameters(SSLParameters sslParameters) {
+      this.sslParameters = sslParameters;
+      return this;
+    }
+
+    public Builder withHostnameVerifier(HostnameVerifier hostnameVerifier) {
+      this.hostnameVerifier = hostnameVerifier;
+      return this;
+    }
+
+    public Builder withHostAndPortMapper(HostAndPortMapper hostAndPortMapper) {
+      this.hostAndPortMapper = hostAndPortMapper;
+      return this;
+    }
+
+    public int getConnectionTimeout() {
+      return connectionTimeout;
+    }
+
+    public int getSoTimeout() {
+      return soTimeout;
+    }
+
+    public boolean isSsl() {
+      return ssl;
+    }
+
+    public SSLSocketFactory getSslSocketFactory() {
+      return sslSocketFactory;
+    }
+
+    public SSLParameters getSslParameters() {
+      return sslParameters;
+    }
+
+    public HostnameVerifier getHostnameVerifier() {
+      return hostnameVerifier;
+    }
+
+    public HostAndPortMapper getHostAndPortMapper() {
+      return hostAndPortMapper;
+    }
+  }
```

Comment:
Should we introduce lombok here and avoid all this boiler plate code? https://projectlombok.org/

Reply:
Would it add an extra dependency?
- If yes, then, no we won't use that.
- If no, then, may be you can help us to add it after this PR gets approved.

Reply:
Yes, and no. It's an extra dependency, but it's a compile-only dependency so it doesn't propagate to consumers in any mean, it just runs during compilation and does its magic.

Will see if I can add it in a separate PR :)